src/Scaler/Services/ScalerMetadata.cs

```diff
@@ -46,7 +62,19 @@ private string LeaseAccountHost
 
         public static ScalerMetadata Create(ScaledObjectRef scaledObjectRef)
         {
-            return JsonConvert.DeserializeObject<ScalerMetadata>(scaledObjectRef.ScalerMetadata.ToString());
+            var metadata = JsonConvert.DeserializeObject<ScalerMetadata>(scaledObjectRef.ScalerMetadata.ToString());
+
+            if (string.IsNullOrEmpty(metadata.Connection) && string.IsNullOrEmpty(metadata.ConnectionFromEnv))
+            {
+                throw new JsonSerializationException("Required property 'Connection' not found in JSON.");
```

Comment:
If we are expecting one of Connection and ConnectionFromEnv to not be empty, error message should be "Required property 'Connection' or 'ConnectionFromEnv' not found in JSON." right?

Reply:
Updated.